### PR TITLE
normalization base project / docker image setup

### PR DESCRIPTION
**Pull request by cgardens**
## What
* Base that sets up project structure and docker container for the normalization stuff. Want to get this merged so we can work off the same stuff. This doesn't actually work yet, but it is also not hooked up to anything.
* @ChristopheDuong here is a base for us to work off of as we work on the normalization. For your prototype you can add it here if that is helpful or you can developer it separately. What I've done here is given you a module to write the json schema => dbt model code and got the basic docker image going.

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    *)
+      error "Unknown option: $1"
+      ;;
+    esac
+  done
+
+  case "$CMD" in
+  run)
+    transform-config --config "$CONFIG_FILE" --integration-type "$INTEGRATION_TYPE" --out "$TRANSFORMED_CONFIG"
+#    transform-catalog --catalog "$CATALOG_FILE" --out "$DBT_MODEL"
```
Comment (cgardens) [inline airbyte-integrations/bases/base-normalization/entrypoint.sh:45]: will be filled in more as we write each of these steps.

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import os
+
+print(os)
```
Comment (cgardens) [inline airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py:27]: to handle this bug: https://github.com/airbytehq/airbyte/issues/776. will be removed once we have other imports.

File: airbyte-integrations/bases/base-normalization/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+FROM fishtownanalytics/dbt:0.18.1
```
Comment (michel-tricot) [inline airbyte-integrations/bases/base-normalization/Dockerfile:1]: Should we use the `airbyte/base-airbyte-protocol-python` and do a pip install of dbt instead?

File: airbyte-integrations/bases/base-normalization/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+TRANSFORMED_CONFIG=transformed_config.json
+DBT_MODEL=dbt_model.json
+set -e
```
Comment (michel-tricot) [inline airbyte-integrations/bases/base-normalization/entrypoint.sh:5]: NIT: put this after the shebang, you never know if someone will change how variable are define and you want to make sure the script fails if it fails

File: airbyte-integrations/bases/base-normalization/.dockerignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+*
```
Comment (michel-tricot) [inline airbyte-integrations/bases/base-normalization/.dockerignore:1]: can you just ignore build?

**Review by michel-tricot — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py (hunk 2 above)
Comment (sherifnada) [reply airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py:27]: can you add as a TODO with a link to the GH issue

File: airbyte-integrations/bases/base-normalization/.dockerignore (hunk 5 above)
Comment (cgardens) [reply airbyte-integrations/bases/base-normalization/.dockerignore:1]: sure. just posted in #dev but i thought what i had done here was our preferred best practice, but happy to just ignore `build`

File: airbyte-integrations/bases/base-normalization/Dockerfile (hunk 3 above)
Comment (cgardens) [reply airbyte-integrations/bases/base-normalization/Dockerfile:1]: fair enough.

**Pull request merged**